### ISSUE TITLE
Unable to mock result of IAsyncEnumerable from DbSet using NSubstitute

### ISSUE THREAD

**Issue by krilllind** [romantitov/MockQueryable#49]
Mocking an `IAsyncEnumerable` result from a `DbSet` using NSubstitute does not work and instead yield an empty result set.
A fix for this exact same issue was adressed in #25 but only for the Moq library.
Same override should be applied when using NSubstitute.

The following code snipped should return all users from a DbSet
```csharp
List<UserEntity> users = new List<UserEntity> { new UserEntity() };
DbSet<UserEntity> mockDbSet = users.AsQueryable().BuildMockDbSet();
dbContext.Users.Returns(mockDbSet);

List<UserEntity> result = await userRepository.GetAllUsersAsync().ToListAsync(cancellationToken);

Assert.AreEqual(users.Count, result.Count);
```
Instead, an empty collection is returned.

**Comment by romantitov** [comment]
Thank you  @krilllind for the issue and especially for the PR. I'll review it and make changes available in a new release.

**Comment by romantitov** [comment]
The issue fixed in release 5.0.1
